QuestionAnswering/QA_ICL/data_generation/README.md

```diff
@@ -1,90 +1,155 @@
+Most scripts in this directory are in Python. The command line arguments supported by a script can be enquired by running it with the `-h` flag, e.g. `python lexicion/lexicon_from_label.py -h`. Below, we give an overview of the available scripts.
+
+### Lexicon Generation
+
+- Generate a lexicon for all entities of a base class using only `rdfs:label` data: [lexicon/lexicon_from_label.py](lexicon/lexicon_from_label.py).
 
-### Generation of lexicons
+- Generate a lexicon for all classes that inherit from a base class using only `rdfs:label` data: [lexicon/lexicon_from_owl.py](lexicon/lexicon_from_owl.py).
 
-- Generate lexicon for all entities of a base class using only `rdfs:label` data: [lexicon/lexicon_from_label.py](lexicon/lexicon_from_label.py).
-- Generate lexicon for all classes that inherit from a base class using only `rdfs:label` data: [lexicon/lexicon_from_owl.py](lexicon/lexicon_from_owl.py).
-- Synthesise lexicon with LLM:
+- Synthesise a lexicon with LLM:
   - Optionally, retrieve all literal data associated with entities of a class first: [lexicon/retrieve_literal_data.py](lexicon/retrieve_literal_data.py).
   - Then, run [lexicon/lexicon_from_llm.py](lexicon/lexicon_from_llm.py).
-- Generate lexicon with scripts specific to entity types:
+
+- Generate lexicons specific to an entity type:
   - Entities of `purl:Element` type: [lexicon/Element_lexicon.py](lexicon/Element_lexicon.py).
   - Entities of `disp:Ship` type: [lexicon/Ship_lexicon.py](lexicon/Ship_lexicon.py).
 
-### Generation of KG schema extracts
+### KG Schema Extraction
+
+- Extract edge type info from an ABox exposed via a SPARQL endpoint: [simplified_schema/extract_edgetypes_from_tbox.py](simplified_schema/extract_edgetypes_from_abox.py).
+- Extract relation type info from OWL files: [simplified_schema/extract_relations_from_tbox.py](simplified_schema/extract_schema_from_tbox.py).
+- Extract relation type info from an ABox exposed via a SPARQL endpoint: [simplified_schema/extract_relations_from_abox.py](simplified_schema/extract_schema_from_abox.py)
 
-- Generate an extract of edge types from an ABox exposed via a SPARQL endpoint: [simplified_schema/extract_edgetypes_from_tbox.py](simplified_schema/extract_edgetypes_from_abox.py).
-- Generate an extract of relation types from OWL files: [simplified_schema/extract_relations_from_tbox.py](simplified_schema/extract_schema_from_tbox.py).
-- Generate an extract of relation types from an ABox exposed via a SPARQL endpoint: [simplified_schema/extract_relations_from_abox.py](simplified_schema/extract_schema_from_abox.py)
+### CSV-to-JSON Conversion of Data Request Generation Examples
 
-### Augmentation of data request examples
+[nlq2datareq_examples/csv2json.py](nlq2datareq_examples/csv2json.py)
 
-Augmentation refers to the process of artificially synthesising new data from existing data. 
+### Augmentation of Data Request Generation
 
 - With an OpenAI-compatible API: [make_examples.py](make_examples.py).
```

Comment:
I think the link is not correct? Unsure which script I should run

Reply:
I have updated the link.